### ISSUE TITLE
Add "Grain merge" texture modifier.

### ISSUE THREAD

**Issue by CandyFiend** [minetest/minetest#15259]
### Problem

Can you add a texture modifier that works similarly to the "grain merge" layer mode in gimp? 
When used correctly, this feature will save memory on textures, and mod authors can use original templates for their blocks without having to draw textures.
It should work like this: the artist creates the texture of some kind of node. Next, he decomposes it into wavelets and gets different layers of detail and a layer of color. We don't need a color layer, but we combine layers with texture relief into one layer. If necessary, it can be discolored. Adding this texture to the mod folder. Next, the author of another mod wants to create the same block, but in a different color. It takes a texture (solid fill with color or with the addition of spots) and applies a texture pattern to it in grain merge mode. Thus, he will receive exactly the same block, but for example, not red, but green. And it will be very different from a simple repainting, because the color pattern applied to the texture will be preserved.
I hope you understood what I was trying to explain.

### Solutions

Add the texture modifier "Grain Merge".

### Alternatives

There are no alternatives. Only the lack of such a useful feature.

### Additional context

1) We take the original texture
![1](https://github.com/user-attachments/assets/d3ec76aa-29d3-48f5-8311-68c63dc04900)
2) Separating the detail from the color
![2](https://github.com/user-attachments/assets/c26940d7-9e13-41df-a84a-1c8de113127d)
3) We take a texture with a color
![3](https://github.com/user-attachments/assets/a5eb56bc-dee8-4749-b278-f76df72ea11a)
4) We apply detail to the color using the "merge grain" mode
![4](https://github.com/user-attachments/assets/5387b827-4a26-498d-9041-d19c092d8cd5)
5) The final result
![5](https://github.com/user-attachments/assets/5f326871-89a6-4409-93f2-a5f69c2b942a)

If you take the texture of sand instead of light and apply the detailing of stone bricks or stoves on it, you can get interesting effects.
![6](https://github.com/user-attachments/assets/cd0718bc-5587-4b20-84fc-a77c3374fce4)

**Comment by appgurueu** [comment]
See: https://docs.gimp.org/2.8/en/gimp-concepts-layer-modes.html, "grain merge". This basically just adds the pixel colors and subtracts 128 if I understand it correctly, so it would be fairly easy to implement.

> When used correctly, this feature will save memory on textures

It will not, since the generated textures have to be stored just as well. But it may save media storage / transfer.

> Thus, he will receive exactly the same block, but for example, not red, but green. And it will be very different from a simple repainting, because the color pattern applied to the texture will be preserved.

There are other texture modifiers that roughly fit this use case, such as `[multiply`. Have you thoroughly considered them and if so, why are they no options?

**Comment by CandyFiend** [comment]
> See: https://docs.gimp.org/2.8/en/gimp-concepts-layer-modes.html, "grain merge". This basically just adds the pixel colors and subtracts 128 if I understand it correctly, so it would be fairly easy to implement.
> 
> > When used correctly, this feature will save memory on textures
> 
> It will not, since the generated textures have to be stored just as well. But it may save media storage / transfer.
> 
> > Thus, he will receive exactly the same block, but for example, not red, but green. And it will be very different from a simple repainting, because the color pattern applied to the texture will be preserved.
> 
> There are other texture modifiers that roughly fit this use case, such as `[multiply`. Have you thoroughly considered them and if so, why are they no options?

Yes, speaking about saving memory, I meant storing and transferring media. It is easier to send 16 color textures and 4 patterns (cobblestone, stone, brick, polished stone) than 4x16 textures. And if desired, the color textures can be replaced with white and simply repainted.

Multiply (this is clearly not what you would expect to see when transferring a pattern to a color base)
![7](https://github.com/user-attachments/assets/ede493d1-5d63-475f-8863-05cd3008fb05)
